src/app/reserve_queue/reserve_queue.cpp

```diff
@@ -0,0 +1,95 @@
+#include "reserve_queue.hpp"
+#include <algorithm>
+
+namespace
+{
+bool addPatientIfNotPresent(ReserveQueue::PatientQueueType& patients_queued, const std::shared_ptr<Patient>& patient)
+{
+    auto patient_iterator = std::ranges::find(patients_queued, patient);
+
+    if (patient_iterator == patients_queued.end())
+    {
+        patients_queued.emplace_back(patient);
+
+        return true;
+    }
+
+    return false;
+}
+
+void removePatientFromQueue(ReserveQueue::PatientQueueType& patients_queued, const std::shared_ptr<Patient>& patient)
+{
+    if (auto patientSearched = std::ranges::find(patients_queued, patient); patientSearched != patients_queued.end())
+    {
+        patients_queued.erase(patientSearched);
+    }
+}
+} // namespace
+
+bool ReserveQueue::addAwaitingPatient(Treatment treatment, const std::shared_ptr<Patient>& patient)
```

Comment:
it can be const Treatment& 
other places also

Reply:
I do not agree with this one.
Treatment is an enum type. const ref& might be an overkill and less readable.
Thus is not recommended in Cpp guidelines (somewhere)

Reply:
Can't find it also, but okay, readibility convinces me :D